engine/api/mailer/Mailer.php

```diff
@@ -124,6 +124,28 @@ static function createEventPublicationEmail(Event $event)
         Le Nose<br>
         <a href = 'www.nose42.fr' >www.nose42.fr</a>";
         $mailer = Mailer::create();
-        return $mailer->createEmail(Mailer::getGlobalAddress(), $subject, $content);
+        if (!$event->groups->isEmpty()) {
+            $users = [];
+            foreach ($event->groups as $group) {
+                foreach ($group->members as $user) {
+                    if ($user->real_email) {
+                        $users[$user->real_email] = ($user->first_name . " " . $user->last_name) ?? '';
+                    }
+                }
+            }
+            if (!empty($users)) {
+                $mailer->createBulkEmails($users, $subject, $content);
+            }
+            return $mailer;
+        } else {
+            //send email to all the club by default
+            $users = [];
+            foreach (UserService::getAll(DB::getInstance()) as $user) {
```

Comment:
This part would deserve a separate DQL query too along with a service method, especially since we would probably want to implement an email preference mechanism soon